src/avram/migrator/drop_index_statement.cr

```diff
@@ -22,14 +22,14 @@ class Avram::Migrator::DropIndexStatement
 
   ALLOWED_ON_DELETE_STRATEGIES = %i[cascade restrict]
 
-  def initialize(@table : Symbol, @columns : Columns, @if_exists = false, @on_delete = :do_nothing)
+  def initialize(@table : Symbol, @columns : Columns | Nil = nil, @if_exists = false, @on_delete = :do_nothing, @name : String? | Symbol? = nil)
   end
```

Comment:
```suggestion
  @columns : Columns?
  @name : Symbol | String | Nil
  
  def initialize(@table : Symbol, @columns : Columns, @if_exists = false, @on_delete = :do_nothing)
  end
  
  def initialize(@table : Symbol, @name : String | Symbol, @if_exists = false, @on_delete = :do_nothing)
  end
```

This makes it so you can't leave both column and string blank, and makes it clear only one should be used but not both

Reply:
@paulcsmith Hm. Good point. Looks great. But there is some ambiguity.
`alias Columns = Symbol | Array(Symbol)`

If I send the parameter as a String, I get `@name`. But if I send the parameter as a Symbol, I get `@columns`. I think this can be confusing. 

I'm not sure how to do the right thing. What do you think?